eZ/Publish/Core/Repository/User/PasswordValidator.php

```diff
@@ -8,24 +8,109 @@
 
 namespace eZ\Publish\Core\Repository\User;
 
+use DateInterval;
+use DateTime;
+use DateTimeImmutable;
 use eZ\Publish\API\Repository\Values\ContentType\FieldDefinition;
+use eZ\Publish\API\Repository\Values\User\PasswordInfo;
+use eZ\Publish\API\Repository\Values\User\User;
+use eZ\Publish\API\Repository\Values\User\User as APIUser;
+use eZ\Publish\Core\FieldType\User\Type as UserType;
+use eZ\Publish\Core\FieldType\ValidationError;
 use eZ\Publish\Core\Repository\Validator\UserPasswordValidator;
 
 /**
  * @internal
  */
 final class PasswordValidator implements PasswordValidatorInterface
 {
+    /** @var \eZ\Publish\API\Repository\PasswordHashService */
+    private $passwordHashService;
+
+    public function __construct(PasswordHashService $passwordHashService)
+    {
+        $this->passwordHashService = $passwordHashService;
+    }
+
     /**
      * @return \eZ\Publish\SPI\FieldType\ValidationError[]
      */
-    public function validatePassword(string $password, FieldDefinition $userFieldDefinition): array
-    {
+    public function validatePassword(
+        string $password,
+        FieldDefinition $userFieldDefinition,
+        ?User $user = null
+    ): array {
         $configuration = $userFieldDefinition->getValidatorConfiguration();
         if (!isset($configuration['PasswordValueValidator'])) {
             return [];
```

Comment:
This is existing code and off topic, but still: If no validator configuration is set, it returns empty array, i.e. no errors, the password is considered valid for storage. That's logical, but is there any actual use case (outside of dev) where it makes sense to have zero password requirements? Sure, our password validation UI may allow disabling all validation, but should it? Or should we set some absolute minimum requirement, like length? 8 chars is the current NIST minimum: https://en.wikipedia.org/wiki/Password_policy

The industry has moved away from mandating detailed requirements, but some minimum requirements are still considered sensible. My guess is some clients may not like that's possible to configure it into an unsafe state. We could allow it, but show a warning message if you do: "This is unsafe, are you sure?" Of course it will always be possible to configure _some_ aspect of the system stack into an unsafe state, that's unavoidable. Anyway, it doesn't belong in this PR, but it would be nice to hear what others think.

Reply:
We provide good zero-config - there are password rules required which can be disabled. We shouldn't cripple flexibility of the system. If you decide to use no requirements - you should be allowed at your own risk.